The build times org.apache.http package cannot be found

JingFengWang opened the pull request:
…org.apache.http package cannot be found

### Change Logs
Add http dependency in ../hudi-common/pom.xml

```html
<dependencies>
    ...
    <dependency>
      <groupId>org.apache.httpcomponents</groupId>
      <artifactId>httpcore</artifactId>
    </dependency>
    ...
</dependencies>
```

### Impact
node


### Risk level (write none, low medium or high below)
low


### Documentation Update
none

### Contributor's checklist

- [ ] Read through [contributor's guide](https://hudi.apache.org/contribute/how-to-contribute)
- [ ] Change Logs and Impact were stated clearly
- [ ] Adequate tests were added if applicable
- [ ] CI passed

danny0405 commented:
Can you show us the dependency tree, which jar/module is dependent on the `httpcore` jar?

JingFengWang commented:
[INFO] --------------------< org.apache.hudi:hudi-common >---------------------
[INFO] Building **hudi-common** 0.13.1                                       [3/50]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ hudi-common ---
[INFO] org.apache.hudi:hudi-common:jar:0.13.1
[INFO] +- org.openjdk.jol:jol-core:jar:0.16:compile
[INFO] +- org.apache.logging.log4j:log4j-1.2-api:jar:2.17.2:provided
[INFO] +- com.lmax:disruptor:jar:3.4.2:compile
[INFO] \- **org.apache.httpcomponents:httpcore:jar:4.4.1:compile**

> Can you show us the dependency tree, which jar/module is dependent on the `httpcore` jar?

JingFengWang commented:
> Can you show us the dependency tree, which jar/module is dependent on the `httpcore` jar?

[INFO] --------------------< org.apache.hudi:hudi-common >---------------------
[INFO] Building hudi-common 0.13.1                                       [3/50]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ hudi-common ---
[INFO] org.apache.hudi:hudi-common:jar:0.13.1
[INFO] +- org.openjdk.jol:jol-core:jar:0.16:compile
[INFO] +- org.apache.logging.log4j:log4j-1.2-api:jar:2.17.2:provided
[INFO] |  \- org.apache.logging.log4j:log4j-api:jar:2.17.2:provided
[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.6.7:compile
......
[INFO] +- org.lz4:lz4-java:jar:1.8.0:compile
[INFO] +- joda-time:joda-time:jar:2.9.9:test
[INFO] +- com.lmax:disruptor:jar:3.4.2:compile
[INFO] \- org.apache.httpcomponents:httpcore:jar:4.4.1:compile

danny0405 commented:
My local build works as expected, not sure what issue you have encountered.

JingFengWang commented:
> My local build works as expected, not sure what issue you have encountered.

![CAPTURE_2023728_94503](https://github.com/apache/hudi/assets/18161501/9cb9dff6-64e9-4c1d-9149-a02792267253)

danny0405 commented:
Did you check your local mvn repository for the http-core jar, maybe you can just remove the local jar and let the mvn to re-download it then retry the build.

JingFengWang commented:
> Did you check your local mvn repository for the http-core jar, maybe you can just remove the local jar and let the mvn to re-download it then retry the build.

I changed the versions of maven and scala, using 3.8.4 and 2.12.10 respectively, and the original 3.8.7 and 2.11.12, and deleted the ./m2/repository file. Recompiled hudi successfully. The phenomenon I found is that the dependency path and scope of httpcore in the hudi-common module have changed, and the scope has changed from test to compile.

![CAPTURE_2023728_155650](https://github.com/apache/hudi/assets/18161501/94ceea89-696e-475d-abb4-98bb9c214c5c)

**Thank you very much for pointing me in the direction to solve the problem.**
